blockprod/src/detail/block_maker.rs

```diff
@@ -185,3 +185,466 @@ impl BlockMaker {
         Ok(())
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use crate::tests::setup_blockprod_test;
+    use chainstate_types::{BlockIndex, BlockPreconnectData, ConsensusExtraData};
+    use crypto::random::make_pseudo_rng;
+    use mempool::{MempoolInterface, MempoolSubsystemInterface};
+    use mocks::{MempoolInterfaceMock, MockChainstateInterfaceMock};
+    use std::sync::atomic::Ordering::Relaxed;
+    use subsystem::CallRequest;
+
+    use chainstate::{
+        chainstate_interface::ChainstateInterface,
+        BlockError::{self, PrevBlockNotFound},
+        ChainstateError,
+    };
+
+    use common::{
+        chain::{
+            block::{timestamp::BlockTimestamp, BlockReward, ConsensusData},
+            Block,
+        },
+        primitives::{BlockHeight, Id, H256},
+        time_getter::TimeGetter,
+        Uint256,
+    };
+
+    use super::*;
+
+    #[tokio::test]
+    async fn collect_transactions_subsystem_error() {
+        let (mut manager, chain_config, chainstate, _mempool) = setup_blockprod_test();
+
+        let mock_mempool = MempoolInterfaceMock::new();
+
+        let mock_mempool_subsystem = manager.add_subsystem_with_custom_eventloop("mock-mempool", {
+            let mock_mempool = mock_mempool.clone();
+            move |call: CallRequest<dyn MempoolInterface>, shutdn| async move {
+                mock_mempool.run(call, shutdn).await;
+            }
+        });
+
+        mock_mempool_subsystem.call_mut({
+            let shutdown = manager.make_shutdown_trigger();
+            move |_| shutdown.initiate()
+        });
```

Comment:
Do you need `call_mut` here, as opposed to plain `call`? It does not touch the mempool object

Reply:
Ah cool, I've updated everywhere it's a `call_mut()` when it didn't need to be